src/io/path_manager.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cstdint>
+#include <filesystem>
+
+namespace openeaw::io {
+
+/// @brief enum containing all the possible installation types
+enum InstallationTypes : std::uint8_t
+{
+    steam,
+    maxinstalltypes
```

Comment:
Please no 'max' or 'count' or anything like that. Just iterate over an explicit list of types when needed.